src/components/DailyVideo.tsx

```diff
@@ -182,14 +182,19 @@ export const DailyVideo = forwardRef<HTMLVideoElement, Props>(
               });
             }
           });
-        };
+        }
 
         handleResize();
-        video?.addEventListener('resize', handleResize);
+        video.addEventListener('loadedmetadata', handleResize);
+        video.addEventListener('resize', handleResize);
 
-        return () => video?.removeEventListener('resize', handleResize);
+        return () => {
+          if (frame) cancelAnimationFrame(frame);
```

Comment:
I’m not sure why all of these `request/cancelAnimationFrame`s are needed, but I assume it’s related to a weird video edge case that I don’t have context on (eg. a video loading in a background tab).

If these are only intended to reorder the React lifecycle, I think moving from a `useEffect` to a callback ref would let you delete these and simplify the code.

In any case, I think adding a `cancelAnimationFrame` here prevents a callback firing on an already unmounted component.